File: app/src/main/java/com/esoxjem/movieguide/listing/ListingModule.java
package com.esoxjem.movieguide.listing;

import com.esoxjem.movieguide.favorites.FavoritesInteractor;
import com.esoxjem.movieguide.network.RequestHandler;
import com.esoxjem.movieguide.listing.sorting.SortingOptionStore;

import dagger.Module;
import dagger.Provides;

/**
 * @author pulkitkumar
 * @author arunsasidharan
 */
@Module
public class ListingModule
{
    @Provides
    MoviesListingInteractor provideMovieListingInteractor(FavoritesInteractor favoritesInteractor,
                                                          RequestHandler requestHandler,
                                                          SortingOptionStore sortingOptionStore)
    {
        return new MoviesListingInteractorImpl(favoritesInteractor, requestHandler, sortingOptionStore);
    }

    @Provides
    MoviesListingPresenter provideMovieListingPresenter(MoviesListingInteractor interactor)
    {
        return new MoviesListingPresenterImpl(interactor);
    }
}

File: app/src/main/java/com/esoxjem/movieguide/movies/listing/ListingModule.kt
package com.esoxjem.movieguide.movies.listing

import com.esoxjem.movieguide.movies.favorites.FavoritesContract
import com.esoxjem.movieguide.movies.sorting.SortingOptionStore
import com.esoxjem.movieguide.network.TmdbWebService
import dagger.Module
import dagger.Provides

/**
 * @author pulkitkumar
 * @author arunsasidharan
 */
@Module
class ListingModule {
    @Provides
    fun provideMovieListingInteractor(favoritesInteractor: FavoritesContract.Interactor,
                                      tmdbWebService: TmdbWebService,
                                      sortingOptionStore: SortingOptionStore): MovieListingContract.Interactor {
        return MoviesListingInteractor(favoritesInteractor, tmdbWebService, sortingOptionStore)
    }

    @Provides
    fun provideMovieListingPresenter(interactor: MovieListingContract.Interactor): MovieListingContract.Presenter {
        return MoviesListingPresenter(interactor)
    }
}
